CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.7)
+cmake_minimum_required(VERSION 3.10)
```

Comment:
Is this version available on all systems we are building the library on?

Reply:
We are following LAMMPS lead here, see https://github.com/lammps/lammps/pull/1723#issuecomment-542280424 for why CMake-3.10 is reasonable beside the fact that it was released in Nov. 2017.

Reply:
`trusty` (Ubuntu 14.04) is EOL April 2022 for example and ships with cmake-2.8.12.2.

Reply:
Ok, just making sure. I don't know what's running on current supercomputers these days :smile:

Reply:
If you want to support `trusty` go for it, but 6 years is a long time ago. I would say if somebody runs `trusty` they need to stick to `bml-1.3.1` instead of the latest master.

Reply:
As I said, I don't know what OSes currently used systems at the labs are running. The last time I checked they were at best "conservative" in their upgrade schedule.

The main point I was trying to make is that we need to agree that we will drop support for CMake < 3.10.

Reply:
@suemni @cnegre your call.